### PR TITLE
added crude multi-threading

### PR DESCRIPTION
**Pull request by AndreasKarasenko**
Adds the option to pass num_workers to predict to speed up inference.
Tested for Zero, Few and Dynamic shot classes.

### CODE REVIEW

File: skllm/models/_base/classifier.py [hunk 1]
```diff
@@ -211,24 +212,30 @@ def fit(
...
         np.ndarray
             The predicted classes as a numpy array.
         """
         X = _to_numpy(X)
-        predictions = []
-        for i in tqdm(range(len(X))):
-            predictions.append(self._predict_single(X[i]))
+        
+        if num_workers > 1:
+            raise DeprecationWarning("Passing num_workers to predict is temporary and will be removed in the future.")
```
Comment (OKUA1) [inline skllm/models/_base/classifier.py:235]: I don't think raising is preferred here as it interrupts the execution. I would suggest using `warnings.warn` instead.

**Review by OKUA1 — CHANGES_REQUESTED** [review]
Hi, please see the comment regarding the usage of the warnings.

File: skllm/models/_base/classifier.py (hunk 1 above)
Comment (AndreasKarasenko) [reply skllm/models/_base/classifier.py]: fixed

**Comment by AndreasKarasenko** [comment]
I'll add some documentation for this next week.

**Pull request merged**